sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Latest.java

```diff
@@ -153,11 +153,12 @@ public T extractOutput(TimestampedValue<T> accumulator) {
    * @param <K> The key type of the elements being combined.
    * @param <V> The value type of the elements being combined.
    */
-  public static <K, V> PTransform<PCollection<KV<K, V>>, PCollection<KV<K, V>>> perKey() {
+  public static <K, V> PerKey<K, V> perKey() {
     return new PerKey<>();
   }
 
-  private static class Globally<T> extends PTransform<PCollection<T>, PCollection<T>> {
+  /** Implementation of {@link #globally()}. */
+  public static class Globally<T> extends PTransform<PCollection<T>, PCollection<T>> {
```

Comment:
I'm not seeing how this discussion is related to that discussion.

It's more internally-evolvable to return PTransform than to return any concrete class.

And the argument about runner intercepting isn't relevant post-FnApi. No runner can inspect the Java classes specifically, only their standard serialized form.